### ISSUE TITLE
False positive :missing-test-assertion for `=` within `testing` within `are`

### ISSUE THREAD

**Issue by vemv** [clj-kondo/clj-kondo#1284]
> **Sponsor**

**version**

`[clj-kondo "2021.04.23"]`

**platform**

JVM

**problem** / **repro**

The following will trigger :missing-test-assertion, even though in fact the underlying deftest will correctly execute its test assertion:

```clj
(deftest are-with-testing
  (are [f] (testing f
             (= "1" (f 1)))
    str
    pr-str))
```

**Comment by vemv** [comment]
Maybe related: https://github.com/clj-kondo/clj-kondo/issues/603

**Comment by vemv** [comment]
Thanks for the triaging :)

As an additional observation, this line should be an actual fault:

```diff
(deftest are-with-testing
  (are [f] (testing f
+            (= "2" (f 1)))
             (= "1" (f 1)))
    str
    pr-str))
```

i.e. only the `=` placed at tail position would be executed as a test.

Personally I wouldn't mind if only the false positive was fixed. I gave a pass to Eastwood this morning making it handle these cases correctly.

i.e. having this in just one linter suffices for my needs

**Comment by borkdude** [comment]
I prefer fixing #603 first so we have a proper expansion of the `are` macro and not some shortcut solution. I will bump the issue for higher priority.

**Comment by borkdude** [comment]
@vemv `clojure.test/are` should now be properly macro-expanded on master

**Comment by vemv** [comment]
Awesome, thanks much! Cool that clj-kondo performs macroexpansions to some degree :)

**Comment by borkdude** [comment]
These expansions can also be written by users as hooks btw.

**Comment by borkdude** [comment]
;-)

https://twitter.com/stuartsierra/status/1398747886796615683
